bridge/bridgeapp/views.py

```diff
@@ -123,3 +124,28 @@ def post(self, request, thread_id, resp_id):
                 Response.objects.filter(id=resp_id).delete()
         # redirect to single thread page
         return redirect('thread', thread_id, 0)
+
+
+def register(request):
+    if request.method == 'POST':
+        form = UserCreationForm(request.POST)
+
+        if form.is_valid():
+            form.save()
+            username = form.cleaned_data['username']
+            password = form.cleaned_data['password1']
+            user = authenticate(username=username, password=password)
+            login(request, user)
+            return redirect('home')
+
+    else:
+        form = UserCreationForm()
+
+    context = {'form': form}
+    return render(request, 'registration/register.html', context)
+
+
+def logout(request):
+    if request.method == 'GET':
```

Comment:
If you make this a POST request you can get two potential benefits
1. Prevent people from creating "logout attacks". Imagine someone on some other site crafts an image targeting your site, `<img src=https://othersite.com/logout.html />`. Just loading the page would trigger the image load which would trigger a logout without user consent. This usee to be a big pain on feed/forum based sites if someone posted something that logged out users from the landing page.
2. If it's a GET request and you put your site behind a CDN, it could cache the request for the user or even globally and show you the logout confirmation page without ever even hitting the server, possibly dropping the cookie header to delete the auth token.

Reply:
My suggestion here isn't around whether there's a separate view to log in or not, just that the request to process the log out should be received via POST.

Reply:
Oh, my bad. I wasn't making comments on your comment. My comments are on the code for a logout functional view. I'll remove the previous and make it on its own.